src/common/containers/Header.jsx

```diff
@@ -50,17 +60,35 @@ export default class Header extends Component {
                         </li>
                     </ul>
                 )}
-
+                    
                 <ul className="order-1 order-md-3 navbar-nav navbar-no-expand ml-auto">
                     <li className="nav-item">
                         <a href="/dashboard" className="nav-link">Dashboard</a>
                     </li>
                     <li className="nav-item">
                         <a href="/blog" className="nav-link">Blog</a>
                     </li>
-                    <li className="nav-item">
-                        <a className="nav-link">{this.state.user.username}</a>
+                    {this.state.userIsAuth?(
+                        <li className="nav-item">
+                            <a href="/singin" className="nav-link">Sing in</a>
+                        </li>       
+                    )
+                    :(
+                    <li className="nav-item dropdown user-menu show" onClick={this.showDropdown}>
+                        <a  className="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="true">
+                        <span className="d-none d-md-inline">{this.state.user.username}</span> </a>
+                        <ul className="dropdown-menu dropdown-menu-lg dropdown-menu-right show" style={this.state.menuOpen ? {'display': 'block'} : {'display': 'none'}}>
+                            <li className="user-header bg-primary">
+                                <img src="" className="img-circle elevation-2" alt="User Image"></img>
+                                <p>{this.state.user.username}</p>
+                            </li>
+                            <li class="user-footer">
+                                <a href="#" className="btn btn-default btn-flat">Profile</a>
+                                <a href="#" className="btn btn-default btn-flat float-right">Sign out</a>
```

Comment:
Let's implement a function in `src/utils/` that will log out the user and redirect them to the main page.